lib/index.d.ts

```diff
@@ -817,6 +828,14 @@ interface NamedSubscription {
     timeToLive: number;
 }
 
+type ExtractPropertyNamesOfType<T, S> = {
+    [K in keyof T]: T[K] extends S ? K : never
+}[keyof T];
+
+type RealmPartialModel<T> =
```

Comment:
I expect to see license fees once that gets merged ;p

Reply:
Of course! And I imagine they'll just go directly out of my salary... 😁